python/GafferSceneUI/LightEditor.py

```diff
@@ -285,6 +286,16 @@ def __transferSelectionFromContext( self ) :
 
 	def __buttonDoubleClick( self, pathListing, event ) :
 
+		# A small corner area below the vertical scroll bar may pass through
+		# to us, causing odd selection behavior. Check that we're within the
+		# scroll area.
+		qPoint = pathListing._qtWidget().viewport().mapFrom(
+			pathListing._qtWidget(),
+			QtCore.QPoint( event.line.p0.x, event.line.p0.y )
+		)
+		if qPoint.x() > pathListing._qtWidget().viewport().size().width() or qPoint.y() > pathListing._qtWidget().viewport().size().height() :
+			return False
```

Comment:
It would be great if we could avoid accessing the Qt internals here - the main idea behind `GafferUI` is that the `Widget` classes hide the "Qtness" of their implementation, so PathListingWidget should be usable from the public API alone. Perhaps we can use `pathListing.pathAt( event ) is None` to determine that the user can't possibly have double-clicked the selection?

A bit more background on this : the need to call `.viewport().mapFrom()` rather than just `.mapFrom()` is pretty unobvious, and has [caused](61bf49158aea5d623d652bc0cceef1da370971ea) [trouble](714c8d8f4a0ec9eaabcd7f89d087faa1e4f17c21) in the past. That's why I'm particularly keen to avoid repetition and rely on public APIs in this instance.

Reply:
That makes total sense, I'm all for avoiding Qt when possible. Fixed in fdaed3f5047094dbb34140b6262f80e32a961a5a